### DISCUSSION TITLE
Is there an API?

### DISCUSSION THREAD

**Discussion by pcuttriss** [padloc/padloc#682] [Q&A] [answered]
Evening folks

Is there an API interface that can be utilised with other code e.g. Python, for automating some aspects e.g. backups?

For example looking at Bitwarden they have an api: https://bitwarden.com/help/public-api/

Is there anything similar?
Is there an intention to do anything similar?

Ta

Paul C

**Comment by MaKleSoft** [comment, marked as answer]
Hello @pcuttriss, and thanks for your question!

No, there is currently no public API, and no plans to add one. Technically, you can use the [internal API](https://github.com/padloc/padloc/blob/main/packages/core/src/api.ts), but that requires authenticating as a user, plus there is a lot of authentication and encryption logic that would have to be ported to be able to use the API in another language.

May I ask what you would like to use a public API for?

**Reply by pcuttriss** [reply]
Morning Martin. 
I was confusing myself with the whole public API piece....what I was after was the internal API to programmatically manage the vault without having to use a browser interface.
Thanks for the link to the internal API though......if I get some time I will take a look at TypeScript but that will have to go onto the list of many many things :-)
Thanks for the reply though.
Paul C
